Update gear.rb

deuber opened the pull request:
Added missing "frame_type".
See https://developers.strava.com/docs/reference/#api-Gears
![Strava-Developers response](https://user-images.githubusercontent.com/11639983/130671526-fd32189a-ec20-4b6a-b1f5-f04b6c9612a8.png)

deuber commented:
hang on, think i see what's needed.

deuber commented:
I downloaded a zip of https://github.com/deuber/strava-ruby-client, unpacked it. Ran bundle install, then ran from within strava-ruby-client-master below, but still auth issues.

`STRAVA_ACCESS_TOKEN=<my access token here>  rspec spec/strava/api/client/endpoints/gears/gear_spec.rb`

![Strava-API token](https://user-images.githubusercontent.com/11639983/131200665-2448b783-30e4-4552-9f2f-7089bbfb256e.png)

Sorry, what am I missing?
Thanks

dblock commented:
You should be cloning the repo with `git` and doing all this command-line. Restart by following https://github.com/dblock/strava-ruby-client/blob/master/CONTRIBUTING.md? You make a fork, check out the code, do `bundle install`, then `rake` to see all tests pass. Then you can make changes that I described above. LMK if you need more help and thanks for hanging in here!

deuber commented:
Thanks for your patience, sorry been a while since done this.

Created clone as directed:
https://github.com/dblock/strava-ruby-client/blob/master/CONTRIBUTING.md

```
git clone https://github.com/contributor/strava-ruby-client.git
cd strava-ruby-client
git remote add upstream https://github.com/dblock/strava-ruby-client.git
```

I ran bundle install without issues, but when I run
`>bundle exec rake`
Running RuboCop...
RuboCop failed!


Guess I need different vers of RuboCop? 

So deleted gemfile.lock and made change to gemfile (below).
  #gem 'rubocop', '~> 0.61.1'
  gem 'rubocop'
  
  Ran bundle install again without errors
  But see this now.
`  >bundle exec rake`
Running RuboCop...
Inspecting 134 files
134 files inspected, 159 offenses detected, 156 offenses auto-correctable

Tip: Based on detected gems, the following RuboCop extension libraries might be helpful:
  * rubocop-rake (https://github.com/rubocop/rubocop-rake)
  * rubocop-rspec (https://github.com/rubocop/rubocop-rspec)

You can opt out of this message by adding the following to your config (see https://docs.rubocop.org/rubocop/extensions.html#extension-suggestions for more options):
  AllCops:
    SuggestExtensions: false
RuboCop failed!

I recently broke my wrist mountain biking, sorry please forgive typos and spellos. Such a pain typing with one hand, thanks!

dblock commented:
Undo any code changes, run `bundle insall`, then `bundle exec rake`, what's the actual error from `rubocop`? It shouldn't fail and you don't need to change anything.

deuber commented:
Did exactly that, but limited output why failed.
Dies almost silently.

Details below.
I've included my rails, ruby versions below.



strava-ruby-client>bundle install
Fetching gem metadata from http://rubygems.org/.........
Resolving dependencies....
Using rake 13.0.6
Using concurrent-ruby 1.1.9
Using minitest 5.14.4
Using bundler 2.2.26
Using public_suffix 4.0.6
Using colored2 3.1.2
Using zeitwerk 2.4.2
Using open4 1.3.4
Using coderay 1.1.3
Using rexml 3.2.5
Using claide 1.0.3
Using faraday-em_synchrony 1.0.0
Using faraday-excon 1.1.0
Using faraday-httpclient 1.0.1
Using ast 2.4.2
Using faraday-net_http_persistent 1.2.0
Using faraday-patron 1.0.0
Using faraday-rack 1.0.0
Using nap 1.1.0
Using ruby2_keywords 0.0.5
Using rchardet 1.8.0
Using no_proxy_fix 0.1.2
Using faraday-em_http 1.0.0
Using diff-lcs 1.4.4
Using dotenv 2.7.6
Using racc 1.5.2
Using faraday-net_http 1.0.1
Using hashie 4.1.0
Using unicode-display_width 1.4.1
Using method_source 1.0.0
Using multi_xml 0.6.0
Using parallel 1.20.1
Using hashdiff 1.0.1
Using powerpack 0.1.3
Using rainbow 3.0.0
Using rspec-support 3.10.2
Using multipart-post 2.1.1
Using vcr 6.0.0
Using i18n 1.8.10
Using tzinfo 2.0.4
Using jaro_winkler 1.5.4
Using cork 0.3.0
Using crack 0.4.5
Using kramdown 2.3.1
Using polylines 0.4.0
Using git 1.9.1
Using nokogiri 1.12.4 (x86_64-darwin)
Using terminal-table 3.0.1
Using ruby-progressbar 1.11.0
Using faraday 1.7.1
Using rspec-core 3.10.1
Using rspec-expectations 3.10.1
Using rspec-mocks 3.10.2
Using activesupport 6.1.4.1
Using pry 0.14.1
Using kramdown-parser-gfm 1.1.0
Using gpx 1.0.0
Using faraday-http-cache 2.2.0
Using addressable 2.8.0
Using rspec 3.10.0
Using sawyer 0.8.2
Using webmock 3.14.0
Using octokit 4.21.0
Using claide-plugins 0.9.2
Using faraday_middleware 1.1.0
Using parser 3.0.2.0
Using danger 8.3.1
Using rubocop 0.61.1
Using danger-plugin-api 1.0.0
Using strava-ruby-client 0.4.2 from source at `.` and installing its executables
Using danger-toc 0.1.3
Using danger-changelog 0.4.2
Bundle complete! 13 Gemfile dependencies, 72 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
Davids-MacBook-Pro-2:~/strava-ruby-client>bundle exec rake
Running RuboCop...
RuboCop failed!
Davids-MacBook-Pro-2:~/strava-ruby-client>rails -v
Rails 6.1.4.1
Davids-MacBook-Pro-2:~/strava-ruby-client>ruby -v
ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [x86_64-darwin17]
Davids-MacBook-Pro-2:~/strava-ruby-client>

deuber commented:
I see your https://github.com/dblock/strava-ruby-client/blob/master/.gitignore
shows Gemfile.lock included, so mine is created at install.

dblock commented:
Oooh you’re right. We need to check that in and lock rubocop. Run `rubocop -a` and let’s see if that fixes it.

deuber commented:
Hi,

I ran 
>rubocop -a
I see many "Correctable" literal string messages like this:

```
spec/strava/api/client/endpoints/routes/export_route_gpx_spec.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
Followed by
134 files inspected, 151 offenses detected, 148 more offenses can be corrected with `rubocop -A`
```
followed by 

```
spec/support/shared/it_behaves_like_web_client.rb:101:33: C: [Correctable] Style/GlobalStdStream: Use $stdout instead of STDOUT.
      let(:logger) { Logger.new(STDOUT) }
                                ^^^^^^
spec/support/vcr.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
require 'vcr'
^
strava-ruby-client.gemspec:1:1: C: Gemspec/RequiredRubyVersion: required_ruby_version should be specified.
$LOAD_PATH.push File.expand_path('lib', __dir__)
^
strava-ruby-client.gemspec:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
$LOAD_PATH.push File.expand_path('lib', __dir__)
^

```
Running 'bundle exec rake' yields same results.

Can you fork and try this? As i have made no changes, I assume you'll get same results?

deuber commented:
or, i should run
rubocop -A
?

dblock commented:
So rubocop auto-corrects with `-a`, and for everything else you can just update the config with `rubocop --auto-gen-config`. Then `rubocop` should finish clean. 

Btw, if you don't want to deal with this, [last time the build passed](https://travis-ci.org/github/dblock/strava-ruby-client/builds/704986661?utm_source=github_status&utm_medium=notification) [used](https://travis-ci.org/github/dblock/strava-ruby-client/jobs/704986662#L261) rubocop 0.61.1, so you could just lock the version at that and move on.

deuber commented:
Tried 'rubocop --auto-gen-config' too, but same results with minimal message.
 
>bundle exec rake
Running RuboCop...
RuboCop failed!

I am using rubocop 0.61.1, see my full  gemlock file below.
Maybe you diff yours with below? 

Thanks,
David



```
PATH
  remote: .
  specs:
    strava-ruby-client (0.4.2)
      activesupport
      faraday (>= 1.0.0)
      faraday_middleware
      hashie

GEM
  remote: http://rubygems.org/
  specs:
    activesupport (6.1.4.1)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 1.6, < 2)
      minitest (>= 5.1)
      tzinfo (~> 2.0)
      zeitwerk (~> 2.3)
    addressable (2.8.0)
      public_suffix (>= 2.0.2, < 5.0)
    ast (2.4.2)
    claide (1.0.3)
    claide-plugins (0.9.2)
      cork
      nap
      open4 (~> 1.3)
    coderay (1.1.3)
    colored2 (3.1.2)
    concurrent-ruby (1.1.9)
    cork (0.3.0)
      colored2 (~> 3.1)
    crack (0.4.5)
      rexml
    danger (8.3.1)
      claide (~> 1.0)
      claide-plugins (>= 0.9.2)
      colored2 (~> 3.1)
      cork (~> 0.1)
      faraday (>= 0.9.0, < 2.0)
      faraday-http-cache (~> 2.0)
      git (~> 1.7)
      kramdown (~> 2.3)
      kramdown-parser-gfm (~> 1.0)
      no_proxy_fix
      octokit (~> 4.7)
      terminal-table (>= 1, < 4)
    danger-changelog (0.4.2)
      danger-plugin-api (~> 1.0)
    danger-plugin-api (1.0.0)
      danger (> 2.0)
    danger-toc (0.1.3)
      activesupport
      danger-plugin-api (~> 1.0)
      kramdown
    diff-lcs (1.4.4)
    dotenv (2.7.6)
    faraday (1.7.1)
      faraday-em_http (~> 1.0)
      faraday-em_synchrony (~> 1.0)
      faraday-excon (~> 1.1)
      faraday-httpclient (~> 1.0.1)
      faraday-net_http (~> 1.0)
      faraday-net_http_persistent (~> 1.1)
      faraday-patron (~> 1.0)
      faraday-rack (~> 1.0)
      multipart-post (>= 1.2, < 3)
      ruby2_keywords (>= 0.0.4)
    faraday-em_http (1.0.0)
    faraday-em_synchrony (1.0.0)
    faraday-excon (1.1.0)
    faraday-http-cache (2.2.0)
      faraday (>= 0.8)
    faraday-httpclient (1.0.1)
    faraday-net_http (1.0.1)
    faraday-net_http_persistent (1.2.0)
    faraday-patron (1.0.0)
    faraday-rack (1.0.0)
    faraday_middleware (1.1.0)
      faraday (~> 1.0)
    git (1.9.1)
      rchardet (~> 1.8)
    gpx (1.0.0)
      nokogiri (~> 1.7)
      rake
    hashdiff (1.0.1)
    hashie (4.1.0)
    i18n (1.8.10)
      concurrent-ruby (~> 1.0)
    jaro_winkler (1.5.4)
    kramdown (2.3.1)
      rexml
    kramdown-parser-gfm (1.1.0)
      kramdown (~> 2.0)
    method_source (1.0.0)
    minitest (5.14.4)
    multi_xml (0.6.0)
    multipart-post (2.1.1)
    nap (1.1.0)
    no_proxy_fix (0.1.2)
    nokogiri (1.12.4-x86_64-darwin)
      racc (~> 1.4)
    octokit (4.21.0)
      faraday (>= 0.9)
      sawyer (~> 0.8.0, >= 0.5.3)
    open4 (1.3.4)
    parallel (1.20.1)
    parser (3.0.2.0)
      ast (~> 2.4.1)
    polylines (0.4.0)
    powerpack (0.1.3)
    pry (0.14.1)
      coderay (~> 1.1)
      method_source (~> 1.0)
    public_suffix (4.0.6)
    racc (1.5.2)
    rainbow (3.0.0)
    rake (13.0.6)
    rchardet (1.8.0)
    rexml (3.2.5)
    rspec (3.10.0)
      rspec-core (~> 3.10.0)
      rspec-expectations (~> 3.10.0)
      rspec-mocks (~> 3.10.0)
    rspec-core (3.10.1)
      rspec-support (~> 3.10.0)
    rspec-expectations (3.10.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.10.0)
    rspec-mocks (3.10.2)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.10.0)
    rspec-support (3.10.2)
    rubocop (0.61.1)
      jaro_winkler (~> 1.5.1)
      parallel (~> 1.10)
      parser (>= 2.5, != 2.5.1.1)
      powerpack (~> 0.1)
      rainbow (>= 2.2.2, < 4.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (~> 1.4.0)
    ruby-progressbar (1.11.0)
    ruby2_keywords (0.0.5)
    sawyer (0.8.2)
      addressable (>= 2.3.5)
      faraday (> 0.8, < 2.0)
    terminal-table (3.0.1)
      unicode-display_width (>= 1.1.1, < 3)
    tzinfo (2.0.4)
      concurrent-ruby (~> 1.0)
    unicode-display_width (1.4.1)
    vcr (6.0.0)
    webmock (3.14.0)
      addressable (>= 2.8.0)
      crack (>= 0.3.2)
      hashdiff (>= 0.4.0, < 2.0.0)
    zeitwerk (2.4.2)

PLATFORMS
  x86_64-darwin-17

DEPENDENCIES
  danger-changelog (~> 0.4.2)
  danger-toc (~> 0.1.3)
  dotenv
  gpx
  multi_xml
  polylines
  pry
  rake
  rspec
  rubocop (~> 0.61.1)
  strava-ruby-client!
  vcr
  webmock

BUNDLED WITH
   2.2.26

```

deuber commented:
My bundle install shows 'Using rubocop 0.61.1' too.

deuber commented:
Sure wish we had more to see then this when it fails.

bundle exec rake
Running RuboCop...
RuboCop failed!

deuber commented:
i see where once working from your link
ruby/2.5.0
will try that version as i was using
```
ruby -v
ruby 2.7.4p19
```

deuber commented:
still failing with ruby 2.5.0 but more message.

rbenv install 2.5.0
rbenv local 2.5.0

>ruby -v
ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin17]

gem install bundler
bundle install


bundle exec rake
Running RuboCop...
Inspecting 134 files
...........................................................................CCC........................................................

Offenses:

lib/strava/models/mixins/elevation.rb:22:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gm', format('%.1f', total_elevation_gain_in_meters))
                  ^^
lib/strava/models/mixins/elevation.rb:28:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gft', format('%.1f', total_elevation_gain_in_feet))
                  ^^
lib/strava/models/mixins/distance.rb:26:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gmi', format('%.2f', distance_in_miles))
                  ^^
lib/strava/models/mixins/distance.rb:36:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gyd', format('%.1f', distance_in_yards))
                  ^^
lib/strava/models/mixins/distance.rb:42:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gm', format('%d', distance_in_meters))
                  ^^
lib/strava/models/mixins/distance.rb:42:33: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gm', format('%d', distance_in_meters))
                                ^^
lib/strava/models/mixins/distance.rb:52:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gkm', format('%.2f', distance_in_kilometers))
                  ^^
lib/strava/models/mixins/time.rb:67:44: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
            hours.to_i.positive? ? format('%dh', hours) : nil,
                                           ^^
lib/strava/models/mixins/time.rb:68:46: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
            minutes.to_i.positive? ? format('%dm', minutes) : nil,
                                             ^^
lib/strava/models/mixins/time.rb:69:46: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
            seconds.to_i.positive? ? format('%ds', seconds) : nil
                                             ^^

134 files inspected, 10 offenses detected
RuboCop failed!

deuber commented:
i would expect it not fail as it did before? but maybe due to new code added since?

deuber commented:
Ran

STRAVA_ACCESS_TOKEN=myAccessTokenHere rspec spec/strava/api/client/endpoints/gears/gear_spec.rb

 Don't I need to provide client_id too? Not just access token?
 
 It created gear.yml but still auth issues.

 
 ---
http_interactions:
- request:
    method: get
    uri: https://www.strava.com/api/v3/gear/g3423618
    body:
      encoding: US-ASCII
      string: ''
    headers:
      Authorization:
      - Bearer access-token
      Accept:
      - application/json; charset=utf-8
      User-Agent:
      - Strava Ruby Client/0.4.2
      Accept-Encoding:
      - gzip;q=1.0,deflate;q=0.6,identity;q=0.3
  response:
    status:
      code: 401
      message: Unauthorized
    headers:
      Date:
      - Thu, 02 Sep 2021 22:37:35 GMT
      Content-Type:
      - application/json; charset=utf-8
      Transfer-Encoding:
      - chunked
      Connection:
      - keep-alive
      Via:
      - 1.1 linkerd, 1.1 linkerd
      Vary:
      - Origin
      Server:
      - nginx/1.19.5
      Status:
      - 401 Unauthorized
      X-Request-Id:
      - 807fedf6-752d-hide
      Cache-Control:
      - no-cache
      Referrer-Policy:
      - strict-origin-when-cross-origin
      X-Frame-Options:
      - DENY
      X-Xss-Protection:
      - 1; mode=block
      X-Download-Options:
      - noopen
      X-Content-Type-Options:
      - nosniff
      X-Permitted-Cross-Domain-Policies:
      - none
    body:
      encoding: UTF-8
      string: '{"message":"Authorization Error","errors":[{"resource":"Athlete","field":"access_token","code":"invalid"}]}'
  recorded_at: Thu, 02 Sep 2021 22:37:35 GMT
recorded_with: VCR 6.0.0

dblock commented:
Looks like you made progress with rubocop. The version 0.61.1 is right. The error is probably from new code (`rubocop -a` should fix it).

For the token, first make sure it's actually valid. Does this work?

```
curl -H 'Accept: application/json' -H "Authorization: Bearer put-your-token-here" https://www.strava.com/api/v3/gear/g3423618 
```

Then make sure to get rid of the existing gear.yml, or it will just replay/reuse that one, before running `rspec`.

deuber commented:
no success with rubocop -a
>bundle exec rake
Running RuboCop...
Inspecting 134 files
...........................................................................CCC........................................................

Offenses:

lib/strava/models/mixins/elevation.rb:22:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gm', format('%.1f', total_elevation_gain_in_meters))
                  ^^
lib/strava/models/mixins/elevation.rb:28:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gft', format('%.1f', total_elevation_gain_in_feet))
                  ^^
lib/strava/models/mixins/distance.rb:26:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gmi', format('%.2f', distance_in_miles))
                  ^^
lib/strava/models/mixins/distance.rb:36:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gyd', format('%.1f', distance_in_yards))
                  ^^
lib/strava/models/mixins/distance.rb:42:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gm', format('%d', distance_in_meters))
                  ^^
lib/strava/models/mixins/distance.rb:42:33: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gm', format('%d', distance_in_meters))
                                ^^
lib/strava/models/mixins/distance.rb:52:19: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
          format('%gkm', format('%.2f', distance_in_kilometers))
                  ^^
lib/strava/models/mixins/time.rb:67:44: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
            hours.to_i.positive? ? format('%dh', hours) : nil,
                                           ^^
lib/strava/models/mixins/time.rb:68:46: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
            minutes.to_i.positive? ? format('%dm', minutes) : nil,
                                             ^^
lib/strava/models/mixins/time.rb:69:46: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
            seconds.to_i.positive? ? format('%ds', seconds) : nil
                                             ^^

134 files inspected, 10 offenses detected
RuboCop failed!

See https://stackoverflow.com/questions/50631272/rubocop-throwing-issue-on-date-strftime
Accordingly, I should update it.
Removed rubocop (~> 0.61.1) from gem file and deleted gemlock file and run bundle install again.
With rubocop (1.20.0)  bundle exec rake runs, but fails auth issue. 

I will investigate why my token not working later,  i still see Authorization Error.
Thanks for simple curl test. 

Question:
**Ok if i use rubocop 1.20.0?**

Failures:

  1) Strava::Api::Client#gear returns gear
     Failure/Error: raise Strava::Errors::Fault, response_values(env)
     
     Strava::Errors::Fault:
       Authorization Error

dblock commented:
Ignore rubocop for now, you can run tests with `rspec spec`. If that passes, you're good.

Does your token work with `curl`?

deuber commented:
Will get to that next week, thanks

On Fri, Sep 3, 2021 at 7:23 PM Daniel Doubrovkine (dB.) <
***@***.***> wrote:

> Ignore rubocop for now, you can run tests with rspec spec. If that
> passes, you're good.
>
> Does your token work with curl?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dblock/strava-ruby-client/pull/41#issuecomment-912892280>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYZZLY5FWFAPNP4CN7F5DDUAF7KZANCNFSM5CXNQVYA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

deuber commented:
You wrote:
"For the token, first make sure it's actually valid. Does this work?
`curl -H 'Accept: application/json' -H "Authorization: Bearer put-your-token-here" https://www.strava.com/api/v3/gear/g3423618 `
"

I see this would only work if i know the token for the Strava API demo user with gear with ID g3423618 
curl -H 'Accept: application/json' -H "Authorization: Bearer put-your-token-here" https://www.strava.com/api/v3/gear/g3423618 

Both these curl commands below work as expected, as i know my own token and gearID.

`curl -H 'Accept: application/json' -H "Authorization: Bearer put-your-token-here" https://www.strava.com/api/v3/athlete `

AND here a token plus a valid gear ID.
`curl -H 'Accept: application/json' -H "Authorization: Bearer put-your-token-here" https://www.strava.com/api/v3/gear/b2043043`

Seems g3423618 is hard coded in rspec tests? How then do i get this Strava API demo users token?  I don't see it shared here. https://developers.strava.com/docs/reference/

Note once working gear.yml shows:
```
http_interactions:
- request:
    method: get
    uri: https://www.strava.com/api/v3/gear/g3423618
```
How did that work without knowing access token?

Ok, getting closer.
What am i missing?
Thanks!

deuber commented:
Side note, no issues with Ruby 2.7.4 and rubocop (1.20.0) 

New Gemfile:

source 'http://rubygems.org'

gemspec

group :development, :test do
  gem 'danger-changelog', '~> 0.4.2'
  gem 'danger-toc', '~> 0.1.3'
  gem 'dotenv'
  gem 'gpx'
  gem 'multi_xml'
  gem 'polylines'
  gem 'pry'
  gem 'rake'
  gem 'rspec'
  **gem 'rubocop'**
  gem 'vcr'
  gem 'webmock'
end

Remove Gemfile.lock
>bundle install
rubocop (1.20.0) now seen in new Gemfile.lock

Cloned Gemfile had:
gem 'rubocop', '~> 0.61.1'

This likely was my earlier issue with rubocop.

dblock commented:
> Both these curl commands below work as expected, as i know my own token and gearID.

Ok, great.
 
> Seems g3423618 is hard coded in rspec tests? How then do i get this Strava API demo users token?

Short version: don't need to, change the gear ID requested in code first.

Long version: A VCR request is just a recording for a request/response. The first time you make an actual call, it records a K7 with the request/response. The next time(s) it uses the K7 as long as the request matches (or it tries to make an actual request). We do this so that subsequent runs (tests, my machine, someone else's machine) don't make actual API requests - we'd have to share secret tokens which would be bad (TM).

So, since you're re-recording a VCR request and you only have access to your ID, you should also change the gear ID being requested in code, in https://github.com/dblock/strava-ruby-client/blob/master/spec/strava/api/client/endpoints/gears/gear_spec.rb#L19 (and elsewhere it appears), otherwise it will try to request whichever old ID that was in code, and produce access denied since _you_ don't have access to it. This was a gear ID for some shoes I own :)

deuber commented:
i see now, i need to change most all fields within gear_spec.rb to get it to pass and match my bike, but working auth now. will do tmrw and makes sense, thanks.

deuber commented:
Success!

new gear.yml shows:

response:
    status:
      code: 200
      message: OK

      encoding: UTF-8
      string: '{"id":"b2338517","primary":false,"name":"Trek","nickname":"Trek","resource_state":3,"retired":false,"distance":54349,"converted_distance":54.3,"brand_name":"Trek
        ","model_name":"Madrone","frame_type":3,"description":"white","weight":9.1}'
  recorded_at: Wed, 08 Sep 2021 17:27:38 GMT
  
  
 
Followed Contributing.md 
But before i run "git push origin my-feature-branch" i want to see what I'm about to push.
https://stackoverflow.com/questions/3636914/how-can-i-see-what-i-am-about-to-push-with-git

`git diff --stat --cached origin/master`

Output shows many files, but I only want to commit gear.rb, gear_spec.rb  and gear.yml.

Tips please?

dblock commented:
`git log -p` will show you what you've committed

If you're not sure you can do this:

```
git checkout master
git pull upstream master # makes sure you are in sync with upstream (here)
git checkout -b add-gear-field # new branch 
git merge my-feature-branch --squash # merges changes but leaves them uncommitted

# revert anything you didn't intend to commit, git add and git commit what you did
git push origin add-gear-field
```

dblock commented:
Yes you should push to your fork.

https://gist.github.com/Chaser324/ce0505fbed06b947d962 is a good read

deuber commented:
Argh, seems I didn't follow directions earlier in CONTRIBUTING.md.
git clone https://github.com/**contributor**/strava-ruby-client.git
I cloned yours, should have been mine. I'll restart the process.

deuber commented:
Maybe we close this as I just now created new PR, so work from there?
https://github.com/dblock/strava-ruby-client/pull/45
Thanks!

dblock commented:
Merged via https://github.com/dblock/strava-ruby-client/pull/45

@deuber Amazing to see how long you've hung in here to finish this change! Kudos.

deuber commented:
And kudos to you, thanks for your patience!

On Wed, Sep 22, 2021 at 2:31 PM Daniel Doubrovkine (dB.) <
***@***.***> wrote:

> Merged via #45 <https://github.com/dblock/strava-ruby-client/pull/45>
>
> @deuber <https://github.com/deuber> Amazing to see how long you've hung
> in here to finish this change! Kudos.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dblock/strava-ruby-client/pull/41#issuecomment-925345480>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYZZL2FYPL4AMQMXO2HKU3UDJDJRANCNFSM5CXNQVYA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

dblock commented:
If you want another challenge, https://github.com/dblock/strava-ruby-client/issues/42 should be next. We need to get rid of Travis-CI and replace with GitHub actions. You can do all this in your fork and when it's working, PR.
